src/metabase/driver/h2.clj

```diff
@@ -121,29 +121,46 @@
       (when (instance? SessionLocal session)
         (Parser. session)))))
 
-(defn- contains-ddl?
-  [database query]
+(defn- check-disallow-ddl-commands
+  [{:keys [database] {:keys [query]} :native}]
   (when-let [h2-parser (make-h2-parser database)]
-    (try
-      (let [command      (.prepareCommand h2-parser query)
-            command-type (.getCommandType command)]
-        ;; TODO: do we need to handle CommandList?
-        ;; Command types are organized with all DDL commands listed first
-        ;; see https://github.com/h2database/h2database/blob/master/h2/src/main/org/h2/command/CommandInterface.java
-        (< command-type CommandInterface/ALTER_SEQUENCE))
-      ;; if the query is invalid, then it isn't DDL
-      (catch Throwable _ false))))
-
-(defn- check-disallow-ddl-commands [{:keys [database] {:keys [query]} :native}]
-  (when (and query (contains-ddl? database query))
-    (throw (IllegalArgumentException. "DDL commands are not allowed to be used with h2."))))
+    (when-let [command (try (.prepareCommand h2-parser query)
+                            ;; if the query is invalid, errors will get caught later
+                            (catch Throwable _ nil))]
+      (cond
+        ;; if the command is a CommandList, then it is a multi-statement query
+        (= (type command) org.h2.command.CommandList)
+          ;; TODO: support multiple statements while checking all the command types
+        (throw (IllegalArgumentException. "Only a single statement is allowed."))
+          ;; Command types are organized with all DDL commands listed first
+          ;; see https://github.com/h2database/h2database/blob/master/h2/src/main/org/h2/command/CommandInterface.java
+        (< (.getCommandType command) CommandInterface/ALTER_SEQUENCE)
```

Comment:
`<=`?

Reply:
I deleted this whole function, in favour of just checking the user is non-admin. see my comment below